src/ms3_extract.sh

```diff
@@ -3,33 +3,79 @@
 # Compile the C++ program
 g++ -std=c++17 ace.cpp -o ace
 
+# Academic year and semester (replace with your academic year and semester)
+academic_year_semester="23s2"
+
+# Team numbers (replace with your team numbers)
+team_numbers=21
+
+# The source directory where repositories are cloned (replace this with your own repo path)
+source_dir="../sample/repo"
+
+# The destination directory where data will be saved (replace this with your data path)
+data_dir="../sample/data"
+
+# The destination directory where AI code will be extracted (replace this with your code path)
+code_dir="$data_dir/code"
+
+# The destination directory where AI chat will be extracted (replace this with your chat path)
+chat_dir="$data_dir/chat"
+
+# The destination directory where AI test cases will be extracted (replace this with your test path)
+test_dir="$data_dir/test"
+
+# Define the dataset file path (replace with your dataset file path)
+dataset="$data_dir/dataset.csv"
+
 # Define the error log file path (replace with your error log file path)
-error_log="../sample/data/errorlog.txt"
-
-# Loop through all teams, modify team numbers as needed (only 0 is used here for sample)
-for ((i=0; i<=0; i++)); do
-  input="T$(printf "%02d" $i)"
-  output="T$(printf "%02d" $i).txt"
-
-  # Format team number with leading zeros (T01-T09)
-  team_number=$(printf "%02d" $i)
-
-  # Modify team names as needed due to course supporting different environments
-  case "$i" in
-    8)
-      team_name="23s1-java-spa-team-08"
-      ;;
-    18)
-      team_name="23s1-win-spa-team-18"
-      ;;
-    29)
-      team_name="23s1-win-spa-team-29"
-      ;;
-    *)
-      team_name="23s1-cp-spa-team-$team_number"
-      ;;
-  esac
+error_log="$data_dir/error_log.txt"
+
+# Clear the code directory
+if [ -d "$code_dir" ]; then
+    rm -rf "$code_dir"/*
+fi
+
+# Clear the chat directory
+if [ -d "$chat_dir" ]; then
+    rm -rf "$chat_dir"/*
+fi
+
+# Clear the test directory
+if [ -d "$test_dir" ]; then
+    rm -rf "$test_dir"/*
+fi
+
+# Remove the dataset file if exists
+if [ -f "$dataset" ]; then
+    rm "$dataset"
+fi
+
+# Remove the error log file if exists
+if [ -f "$error_log" ]; then
+    rm "$error_log"
+fi
+
+# Loop through all teams
+for team_number in $(seq 1 $team_numbers); do
+  # Format team numbers with leading zeros
+  formatted_team_number=$(printf "%02d" $team_number)
+  output="T$formatted_team_number.txt"
+
+  # Ensure the test directory exists or create it
+  mkdir -p "$test_dir/T$formatted_team_number"
+
+  # Team numbers 08, 18, and 29 have different suffixes (replace with your team names)
+  if [ "$formatted_team_number" == "08" ]; then
+    team_type="java"
```

Comment:
I think we can comment this out, there are no java teams this sem.

Reply:
Yeah we can modify it when we can access other teams' repositories.